proposed/disable-multi-level-lookup-by-default.md

```diff
@@ -0,0 +1,204 @@
+    .NET 5 app via a private install `dotnet.exe`, but running `dotnet --list-runtimes`
+    will not list .NET 5 runtime as available.
+    * It might make sense to include a special message in the output of `dotnet --info`
+    which would effectively warn about this discrepancy. The output of `dotnet --info`
+    is meant to be humanly readable text and it already includes for example links
+    to install more SDKs so adding another message should have minimal risk
+    of breaking anything. `dotnet --list-runtimes` and `dotnet --list-sdks`
+    on the other hand are not formatted for humans and should not include
+    any additional message.
+* Native hosting APIs
+    * All hosting API scenarios have `.runtimeconfig.json` available and so will
+    act as applications - they will react to target framework version
+    as described above.
+    * `hostfxr_get_dotnet_environment_info` and `hostfxr_get_available_sdks`
+    would behave the same as `dotnet --list-runtimes` and `dotnet --list-sdks` -
+    so multi-level lookup is off by default on `hostfxr` .NET 6+.
+    * `hostfxr_resolve_sdk`, `hostfxr_resolve_sdk2` - similar to the enumeration
+    APIs these will use multi-level lookup off by default when running in
+    `hostfxr` .NET 6+.
+    * `nethost`'s `get_hostfxr_path` API to find `hostfxr` does not react
+    to multi-level lookup in any way (effectively disabled always) and as such
+    is not affected by this change. This means that using private install location
+    will always run the `hostfxr` from the private install location, with no way
+    to allow multi-level lookup for `hostfxr` (existing behavior).
+
+## UX changes
+
+We want to inform the user about this change in some way in cases where it's
+permissible to do so. This means additional messages should not be printed when
+running applications since the app may depend on exact output format. As such
+the additional information should only be provided when running CLI commands (SDK).
+
+Host should print out a similar message when enumerating runtimes and SDKs in
+`dotnet --info`. As mentioned above `dotnet --list-runtimes` and `dotnet --list-sdks`
+should not be affected.
+
+The message should only be shown when there's observable difference in behavior.
+For CLI commands that means when running the command with MLL on-by-default
+would end up using different version of SDK then when running it with
+MLL off-by-default. `dotnet --info` should print the message in all cases where
+there's a global install location (on top of the private one which is used
+to run the command).
+
+## Open questions
+
+### New dotnet CLI actions for install/update
+
+This is effectively an extension to the discrepancy between app behavior and
+`dotnet --info` behavior. What is the desired behavior for the new CLI actions -
+this is yet to be determined.
+
+### Way to dismiss informational message about the change
+
+There should be some way to dismiss the information messages printed by
```

Comment:
We could also consider only showing the message for a period of time, so that we don't have to have a file-based scheme.  Imagine something like "Show it for 1 week past installation."

Reply:
I'd be happy with showing a message during previews only.

Reply:
Added both options to the list.